### PR TITLE
NMS-13759: Split asset topology provider docs

### PR DESCRIPTION
**Pull request by mfuhrmann**
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13759

### CODE REVIEW

File: docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+| Description
+| Options
+
+| opennms:asset-topo-create
+| Creates asset topology.
+| (Uses the default settings if a particular setting is not included in the command.) +
+-l, --label : Asset topology label (shows in topology menu) (Default: asset) +
+-i, --providerId : Unique providerId of asset topology (Default: Asset Topology Provider) +
+-f, --filter : Optional node filter (Default: empty filter; in other words, allow all nodes) +
+-a, --assetLayers : Comma-separated list of asset layers (Default: asset-region,asset-building,asset-rack) +
```
Comment (Bonrob2) [inline docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc:26]:
```suggestion
-a, --assetLayers : Comma-separated list of asset layers (Default: asset-region, asset-building, asset-rack) +
```

File: docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+| Removes asset topology.
+| -i, --providerId : Unique providerId of asset topology (Default: asset).
+
+| opennms:asset-topo-list
+| Lists all asset topologies installed.
+| all : display detailed view including --uriParams string
+
+| opennms:asset-topo-regenerate
+| Regenerates the graphs for the given asset topology definition.
+| -i, --providerId : Unique providerId of asset topology to regenerate (Default: asset)
```
Comment (Bonrob2) [inline docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc:42]:
```suggestion
| -i, --providerId : Unique providerId of asset topology to regenerate (Default: asset).
```

File: docs/modules/operation/pages/topology/asset-topology-provider/introduction.adoc [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+The following diagram shows the structure of a topology generated by the above assetLayers property:
+
+image:asset-topology/graphMLtopologyLayers.jpg[Illustrating how node asset entries are interpreted as layers]
+
+In this example, the `region` asset fields for nodes 1, 2, 3, 4 are set to north.
+All of these nodes are in the same north region.
+The `building` asset fields for node 1 and node 2 are set to 21 (both nodes are in building 21), while the `building` asset fields for node 3 and node 4 are set to 22 (both nodes are in building 22).
+
+The Asset Topology Provider generates four linked graphs for this configuration.
+The layer 0 graph is called `asset-region`, the layer 1 graph is called `asset-building`, and the layer 2 graph is called `nodes`.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/topology/asset-topology-provider/introduction.adoc:35]:
```suggestion
The layer 0 graph is called `asset-region`; the layer 1 graph is called `asset-building`; and the layer 2 graph is called `nodes`.
```

File: docs/modules/operation/pages/topology/asset-topology-provider/node-filtering.adoc [hunk 4]
```diff
@@ -0,0 +1,49 @@
+[[node-filtering]]
+== Node filtering
+
+In many cases, it is desirable to control which nodes are included or excluded from a topology.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/topology/asset-topology-provider/node-filtering.adoc:4]:
```suggestion
In many cases, it is desirable to control which nodes a topology includes or excludes.
```

File: docs/modules/operation/pages/topology/asset-topology-provider/view-additional-notes.adoc [hunk 5]
```diff
@@ -0,0 +1,16 @@
+== View the topology
+
+After installing the topology and refreshing your screen, you should see a new topology display option in the {page-component-title} topology page.
+The label field provides the displayed name of this topology.
+
+The label field does not need to be the same as the providerId that the ReST API uses for the installation or removal of a topology.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/topology/asset-topology-provider/view-additional-notes.adoc:6]:
```suggestion
The label field does not need to be the same as the providerId that the REST API uses for the installation or removal of a topology.
```

**Review by Bonrob2 — COMMENTED** [review]
Just a few changes, otherwise looks good.

**Review by Bonrob2 — DISMISSED** [review]
LGTM!

File: docs/modules/operation/pages/topology/asset-topology-provider/configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+[cols="1,3"]
+|===
+| Parameter           | Description
+| providerId         | The unique name of the provider - used as a handle to install and remove the topology.
+| label              | The name that shows up on the topology menu (must be unique).
+| assetLayers        | List of asset layers (in order). See xref:asset-layers[asset-layers].
+| filters            | List of filters to apply. Filters determine the nodes to include in the graph. See xref:node-filtering[node filtering].
+| preferredLayout    | Preferred node layout in generated graphs.
+| breadcrumbStrategy | Breadcrumb strategy used to display breadcrumbs above each graph.
+|===
```
Comment (mmahacek) [inline docs/modules/operation/pages/topology/asset-topology-provider/configuration.adoc:58]:
```suggestion
The configuration file contains each of the graph definitions as properties in the form:

[source, xml]
----
<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<configs>
    <config>
        <label>Asset Topology Provider</label><1>
        <breadcrumb-strategy>SHORTEST_PATH_TO_ROOT</breadcrumb-strategy><2>
        <provider-id>asset</provider-id><3>
        <preferred-layout>Grid Layout</preferred-layout><4>
        <filters><5>
            <filter>asset-region=South</filter>
        </filters>
        <layers><6>
            <layer>asset-region</layer>
            <layer>asset-building</layer>
            <layer>asset-rack</layer>
        </layers>
    </config>
</configs>
----
<1> The name that shows up on the topology menu (must be unique).
<2> Breadcrumb strategy used to display breadcrumbs above each graph.
<3> The unique name of the provider - used as a handle to install and remove the topology.
<4> Preferred node layout in generated graphs.
<5> List of filters to apply. Filters determine the nodes to include in the graph. See xref:operation:topology/asset-topology-provider/node-filtering.adoc[node filtering].
<6> List of asset layers (in order). See xref:operation:topology/asset-topology-provider/asset-layers.adoc[asset layers].
```

File: docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+If you simply type `asset-topology:create`, a default topology with providerId asset is created.
+
+| opennms:asset-topo-remove
+| Removes asset topology.
+| -i, --providerId : Unique providerId of asset topology (Default: asset).
+
+| opennms:asset-topo-list
+| Lists all asset topologies installed.
+| all : display detailed view including --uriParams string
```
Comment (mmahacek) [inline docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc:38]:
```suggestion
| --all : display detailed view including --uriParams string
```

File: docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+| Creates asset topology.
+| (Uses the default settings if a particular setting is not included in the command.) +
+-l, --label : Asset topology label (shows in topology menu) (Default: asset) +
+-i, --providerId : Unique providerId of asset topology (Default: Asset Topology Provider) +
+-f, --filter : Optional node filter (Default: empty filter; in other words, allow all nodes) +
+-a, --assetLayers : Comma-separated list of asset layers (Default: asset-region, asset-building, asset-rack) +
+-p, --preferredLayout : Preferred Layout (Default: Grid Layout) +
+-b, --breadcrumbStrategy : Breadcrumb strategy (Default: SHORTEST_PATH_TO_ROOT) +
+
+If you simply type `asset-topology:create`, a default topology with providerId asset is created.
```
Comment (mmahacek) [inline docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc:30]:
```suggestion
If you simply type `opennms:asset-topo-create`, a topology with default settings is created.
```

File: docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+| opennms:asset-topo-create
+| Creates asset topology.
+| (Uses the default settings if a particular setting is not included in the command.) +
+-l, --label : Asset topology label (shows in topology menu) (Default: asset) +
+-i, --providerId : Unique providerId of asset topology (Default: Asset Topology Provider) +
+-f, --filter : Optional node filter (Default: empty filter; in other words, allow all nodes) +
+-a, --assetLayers : Comma-separated list of asset layers (Default: asset-region, asset-building, asset-rack) +
+-p, --preferredLayout : Preferred Layout (Default: Grid Layout) +
+-b, --breadcrumbStrategy : Breadcrumb strategy (Default: SHORTEST_PATH_TO_ROOT) +
```
Comment (mmahacek) [inline docs/modules/operation/pages/topology/asset-topology-provider/create-asset-based-topologies-karaf.adoc:28]:
```suggestion
-b, --breadcrumbStrategy : Breadcrumb strategy (Default: SHORTEST_PATH_TO_ROOT)
```

**Review by mmahacek — CHANGES_REQUESTED** [review]

**Comment by mfuhrmann** [comment]
@Bonrob2 @mmahacek Thanks for reviewing and improving!

**Comment by mfuhrmann** [comment]
Oh Oh. I am not sure if the rebase did what I wanted...

**Comment by mmahacek** [comment]
😂 Which branch did you rebase against?

**Comment by mfuhrmann** [comment]
@mmahacek `foundation-2021`

**Comment by mmahacek** [comment]
I was being somewhat rhetorical as that's probably what was intended, but why your copy of the repo has all these commits that aren't in the main foundation branch is beyond me.  Given today is a release day, does @RangerRick have any insight?

**Comment by mfuhrmann** [comment]
Don't spend too much time on that. I can create a new PR for this.

**Comment by RangerRick** [comment]
Release day already happened, so that's academic.  :)

The problem appears to be that you rebased against `foundation-2021` from a bit ago, because if I try to merge the latest 2021 to your branch, there's a _lot_ of changes.  Just one conflict though.

You might want to try rebasing again (make sure you do it against `origin/foundation-2021` after doing a `git fetch` or `git pull` just to make sure you're in sync) and see if that helps.

**Comment by mfuhrmann** [comment]
That was fun... NOT ;-)
Will create a new PR. See you later

**Pull request closed without merging**